### ISSUE TITLE
URL not shown in entry table if scheme is not http/https

### ISSUE THREAD

**Issue by apiraino** [keepassxreboot/keepassxc#1424]
A minor glitch. Created an entry in keepassxc for an ftp site:
```
Title: FreeBSD ftp
Username: ftpuser
Password: ******
URL: ftp://ftp.freebsd.org/
```
This item appears with an empty URL on the password listing. Anything but http/https shows a blank URL on the password listing.

## Expected Behavior
Strip spaces from urls when saved, support any protocol scheme

## Current Behavior
Only http/https URLs are shown, url is not stripped

## Steps to Reproduce (for bugs)
1. Start keepassxc, open password database
2. CTRL+N
3. Insert sample data above and save
4. Password listing shows the new item but url field is empty

Operating system: Ubuntu 17.10
Browser: Firefox 59.0b3
Proxy used: YES

## Debug Info
KeePassXC - Version 2.2.4
Revision: 83f1a53

Libraries:
- Qt 5.9.1
- libgcrypt 1.7.8

Operating system: Ubuntu 17.10
CPU architecture: x86_64
Kernel: linux 4.13.0-31-generic

Enabled extensions:
- Auto-Type
- Browser Integration

**Comment by varjolintu** [comment]
Could you provide some more details about this or exact steps to reproduce it? Did you create the entry from the browser extension or it's already created in KeePassXC but not accessed properbly in the extension?

**Comment by apiraino** [comment]
Hi, I've updated my bug report with the steps to reproduce. Please note that the browser extension is not involved (sorry for not being clearer)

**Comment by eNTi** [comment]
When is this fix going to hit stable?

**Comment by TheZ3ro** [comment]
I think 2.3.1

**Comment by tylert** [comment]
This issue is still a problem in 2.3.1.  If the scheme is http:// or https://, the URL displays correctly.  If you omit the scheme, it automatically assumes http://.

I have been adding other schemes such as smtp://, smtps://, ssh://, hostname://, ssid:// and so on for other types I use to prevent them from being automatically assumed to be http://.

**Comment by Trikas77** [comment]
Additionally:
I used for a long time the scheme 'android-package://' for an Android App which provided auto-fill functionality.

**Comment by apiraino** [comment]
I'm following this PR #1768

**Comment by droidmonkey** [comment]
This was fixed in 2.3.4
